angular-ngrx-scss/src/app/auth/redirect/redirect.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { concatMap, filter, tap } from 'rxjs';
+import { AuthService } from '../services/auth.service';
+
+@Component({
+  selector: 'app-redirect',
+  template: `<div>Redirecting...</div>`,
+})
+export class RedirectComponent implements OnInit {
+  constructor(
+    private authService: AuthService,
+    private route: ActivatedRoute,
+    private router: Router,
+  ) {}
+
+  ngOnInit(): void {
+    /**
+     * When we get the code from GitHub, pass it to the authService and redirect the user to the home page
+     */
+    this.route.queryParamMap
```

Comment:
I think I would put this into a RedirectGuard where I'd return `/` or back to `/signin`.

For more on this please see: https://juristr.com/blog/2018/11/better-route-guard-redirects/

Reply:
This is not a request to change though, because putting this into a guard will not render the `Redirecting...` text. I guess we would need a spinner or a loading indicator using a component portal for that.

Reply:
Fair point here. An improvement to this is potentially underway in another PR from Morgan. Will come back to this if / when those changes are approved.